extensions/amp-ad-network-adsense-impl/0.1/responsive-state.js

```diff
@@ -149,6 +184,31 @@ export class ResponsiveState {
     return /** @type {!ResponsiveState} */ (state);
   }
 
+  /**
+   * Convert the element to container width responsive.
+   *
+   * @param {!Element} element
+   * @return {!Promise<?ResponsiveState>} a promise that resolves when any upgrade is complete.
+   * @private
+   */
+  static convertToContainerWidth_(element) {
+    const vsync = Services.vsyncFor(toWin(element.ownerDocument.defaultView));
+
+    return vsync.measurePromise(() => {
+      const width = String(element./*OK*/ parentElement./*OK*/ clientWidth);
+
+      element.setAttribute('height', ADSENSE_RSPV_WHITELISTED_HEIGHT);
```

Comment:
These mutations need to be in a mutate phase. You can use `vsync.mutatePromise` to schedule one.

Reply:
I tried to switch to mutatePromise, but the ad did not resize / load successfully:
https://screenshot.googleplex.com/tygZVEEjVLP

Reply:
As the class replies on alignToViewport at the end to make the CSS adjustments & mutate. We may not need vsync here.

Reply:
Talked offline, vsync added back.